src/main/java/edu/cornell/kfs/pmw/batch/dataaccess/impl/PaymentWorksIsoFipsCountryDaoJdbc.java

```diff
@@ -0,0 +1,86 @@
+        List<PaymentWorksIsoFipsCountryItem> isoToFipsValueDatabaseRows = findIsoToFipsCountryTranslations();
+        
+        if (ObjectUtils.isNotNull(isoToFipsValueDatabaseRows) && !isoToFipsValueDatabaseRows.isEmpty()) {
+            ListIterator <PaymentWorksIsoFipsCountryItem> isoToFipsValueDatabaseRowsIterator =  isoToFipsValueDatabaseRows.listIterator();
+            String isoKey = isoToFipsValueDatabaseRows.get(0).getIsoCountryCode();
+            List<PaymentWorksIsoFipsCountryItem> isoFipsValues = new ArrayList<PaymentWorksIsoFipsCountryItem>();
+            
+            while (isoToFipsValueDatabaseRowsIterator.hasNext()) {
+                PaymentWorksIsoFipsCountryItem currentCountryValue = isoToFipsValueDatabaseRowsIterator.next();
+                if (isoKey.equalsIgnoreCase(currentCountryValue.getIsoCountryCode())) {
+                    isoFipsValues.add(currentCountryValue);
+                }
+                else {
+                    isoToFipsMap.put(isoKey, isoFipsValues);
+                    isoKey = currentCountryValue.getIsoCountryCode();
+                    isoFipsValues = new ArrayList<PaymentWorksIsoFipsCountryItem>();
+                    isoFipsValues.add(currentCountryValue);
+                }
+            }
+            isoToFipsMap.put(isoKey, isoFipsValues);
+        }
+        for (String key : isoToFipsMap.keySet()) {
+            List<PaymentWorksIsoFipsCountryItem> valuesList = isoToFipsMap.get(key);
+            LOG.info("buildIsoToFipsMapFromDatabase: key: " + key);
+            for (PaymentWorksIsoFipsCountryItem countryItem : valuesList) {
+               LOG.info("buildIsoToFipsMapFromDatabase: value(s): " + countryItem.toString());
+            }
+        }
+        return isoToFipsMap;
+    }
+    
+    private List<PaymentWorksIsoFipsCountryItem> findIsoToFipsCountryTranslations() {
+        try {
+            RowMapper<PaymentWorksIsoFipsCountryItem> mapRow = new RowMapper<PaymentWorksIsoFipsCountryItem>() {
+                public PaymentWorksIsoFipsCountryItem mapRow(ResultSet resultSet, int rowNumber) throws SQLException {
+                    PaymentWorksIsoFipsCountryItem isoToFipsCountryItem = 
+                            new PaymentWorksIsoFipsCountryItem(resultSet.getString(ISO_CNTRY_CD_COL), resultSet.getString(ISO_CNTRY_NM_COL), 
+                                                               resultSet.getString(FIPS_CNTRY_CD_COL), resultSet.getString(FIPS_CNTRY_NM_COL));
+                    return isoToFipsCountryItem;
+                }
+            };
+            return this.getJdbcTemplate().query(buildIsoToFipsCountryTranslationsSql(), mapRow);
+        } catch (Exception e) {
+            LOG.info("PaymentWorksIsoFipsCountryDaoJdbc Exception: " + e.getMessage());
+            return null;
+        }
+    }
+    
+    private String buildIsoToFipsCountryTranslationsSql() {
```

Comment:
Do we need all these mappings in memory? This code will be more difficult to maintain than code that is using the object/relational mapping.

Reply:
The data that is received by KFS via the PaymentWorks integration is not validated at all.  So, in the case of countries, validation and conversion from ISO to FIPS needs to be performed for 5 countries for every DV vendor and 6 countries for every PO vendor. I am not sure about RV vendors yet. With that being said, I am concerned about the PaymentWorks batch jobs becoming database bound for all of that validation/data conversion. That is the reason why I chose the read the database once into a Map design.

Reply:
Thank you for explaining this. I am wondering if in this case it might make sense to use a view so that we can have relational mapping defined in ojb and make the code more clean an easy to maintain.

If you decide against a view I think it would be good to have the SQL defined in a constant.

Reply:
Per our offline conversation, leaving the Map implementation and moving the SQL to a constant at the top of the class.